src/main/java/com/getindata/connectors/http/internal/security/SecurityContext.java

```diff
@@ -0,0 +1,184 @@
+
+        try {
+            this.keystore = KeyStore.getInstance(JKS_STORE_TYPE);
+            this.keystore.load(null, storePasswordCharArr);
+            log.info("Created KeyStore for Http Connector security context.");
+        } catch (Exception e) {
+            throw new RuntimeException(
+                "Unable to create KeyStore for Http Connector Security Context.",
+                e
+            );
+        }
+    }
+
+    /**
+     * Creates an instance of {@link SSLContext} backed by {@link KeyStore} from this {@link
+     * SSLContext} instance.
+     *
+     * @param trustManagers {@link TrustManager} that should be used to create {@link SSLContext}
+     * @return new sslContext instance.
+     */
+    public SSLContext getSslContext(TrustManager[] trustManagers) {
+        try {
+
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+            keyManagerFactory.init(this.keystore, this.storePasswordCharArr);
+
+            // populate SSLContext with key manager
+            SSLContext sslCtx = SSLContext.getInstance("TLSv1.2");
+            sslCtx.init(keyManagerFactory.getKeyManagers(), trustManagers, null);
+            return sslCtx;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Creates TrustManagers for given {@link KeyStore} managed by this instance of
+     * {@link SSLContext}. It is important that all keys and certificates should be added
+     * before calling this method. Any key/certificate added after calling this method
+     * will not be visible by previously created TrustManager objects.
+     * @return an array of {@link TrustManager}
+     */
+    public TrustManager[] getTrustManagers() {
+        try {
+            String alg = TrustManagerFactory.getDefaultAlgorithm();
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(alg);
+
+            trustManagerFactory.init(this.keystore);
+            log.info("Created security Trust Managers for Http Connector security context.");
+            return trustManagerFactory.getTrustManagers();
+        } catch (GeneralSecurityException e) {
+            throw new RuntimeException(
+                "Unable to created Trust Managers for Http Connector security context.",
+                e
+            );
+        }
+    }
+
+    /**
+     * Adds certificate to as trusted. Certificate is added only to this Context's {@link KeyStore}
+     * and not for entire JVM.
+     * @param certPath path to certificate that should be added as trusted.
+     */
+    public void addCertToTrustStore(String certPath) {
+
+        log.info("Trying to add certificate to Security Context - " + certPath);
+        try (FileInputStream certInputStream = new FileInputStream(certPath)) {
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
```

Comment:
Don't we want to put `"X.509"` and similar as string constants?

Reply:
Done